src/main/java/bio/terra/app/controller/DatasetsApiController.java

```diff
@@ -176,6 +178,16 @@ public ResponseEntity<JobModel> deleteDataset(@PathVariable("id") UUID id) {
     return jobToResponse(jobService.retrieveJob(jobId, userReq));
   }
 
+  @Override
+  public ResponseEntity<DatasetSummaryModel> patchDataset(
+      UUID id, DatasetPatchRequestModel patchRequest) {
+    AuthenticatedUserRequest userReq = getAuthenticatedInfo();
+    for (IamAction action : datasetService.patchDatasetIamActions(patchRequest)) {
+      iamService.verifyAuthorization(userReq, IamResourceType.DATASET, id.toString(), action);
+    }
```

Comment:
This will make a separate SAM call for each action. It looks TDR doesn't use it yet, but there's a SAM API that returns all actions for a user for a resource (`resourceActionsV2`). (TDR calls this via `IamProviderInterface.hasActions` but the actions aren't returned.) It would be better to only make a single SAM call here, if possible.

Reply:
After looking at the code, my plan is to merge this PR as it stands and handle this consolidation separately in a new ticket since the changes may be a touch involved.  The endpoint will not be called frequently so we won't clobber SAM in the interim.